CMakeLists.txt

```diff
@@ -671,6 +671,11 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build from: Debug Release RelWithDebInfo MinSizeRel Coverage." FORCE)
 endif()
 
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -mcmodel=large")
+#set(CMAKE_HIP_FLAGS_DEBUG "${CMAKE_HIP_FLAGS_DEBUG} -mcmodel=large")
+set(CMAKE_HIP_FLAGS_DEBUG "${CMAKE_HIP_FLAGS_RELEASE}")
+set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "${CMAKE_SHARED_LINKER_FLAGS} -mcmodel=large")
+
```

Comment:
@dmikushin Wrt https://github.com/pytorch/pytorch/pull/82328#issuecomment-1216232248, it seems this PR specifies the compiler flags for non-ROCm debug builds as well. Is that necessary? I'm not sure if there is a drawback or penalty when using this flag during compilation eg. higher compilation time or larger code objects, but we'd need to justify if this flag will have any negative impact whatsoever when building debug builds for non-ROCm environments.